### PR TITLE
Feature/anmodninger via frontend

### PR DESCRIPTION
**Pull request by mimse**
image

## References

- https://github.com/Energinet-DataHub/team-phoenix/issues/73

### CODE REVIEW

**Review by defectiveAi — DISMISSED** [review]

File: libs/dh/wholesale/feature-request-calculation/src/lib/dh-wholesale-request-calculation.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+<watt-card vater inset="0" *transloco="let t; read: 'wholesale.requestCalculation'">
+  <form [formGroup]="form" (ngSubmit)="requestCalculation()">
+    <vater-flex fill="vertical" gap="m">
+      <vater-stack align="flex-start" direction="column" gap="s">
+        <watt-dropdown
+          translate="wholesale.requestCalculation.processTypes"
+          [label]="t('processType')"
+          [formControl]="form.controls.processType"
+          [options]="progressTypeOptions"
+          dhDropdownTranslator
```
Comment (dzhavat) [inline libs/dh/wholesale/feature-request-calculation/src/lib/dh-wholesale-request-calculation.html:26]: issue: missing translations for some options

image

File: libs/dh/wholesale/feature-request-calculation/src/lib/dh-wholesale-request-calculation.ts [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+  selectedActorQuery = this._apollo.watchQuery({
+    useInitialLoading: true,
+    notifyOnNetworkStatusChange: true,
+    query: graphql.GetSelectedActorDocument,
+  });
+
+  energySupplierQuery = this._apollo.watchQuery({
+    useInitialLoading: true,
+    notifyOnNetworkStatusChange: true,
+    query: graphql.GetActorsForRequestCalculationDocument,
```
Comment (dzhavat) [inline libs/dh/wholesale/feature-request-calculation/src/lib/dh-wholesale-request-calculation.ts:156]: suggestion: add mocks to `GetActorsForRequestCalculation` and `getSelectedActor` queries

image

File: .github/workflows/production-dependencies-license-check.yml [hunk 3]
```diff
@@ -51,7 +51,7 @@ jobs:
       - name: Run license-checker
         id: licenseChecker
         if: ${{ steps.dependencies_have_changed.outputs.dependencies-have-changed == 'true' }}
-        run: echo DEPENDENCIES=$(npx license-checker-rseidelsohn@4 --direct --production --onlyAllow="${{ steps.allowed-licenses.outputs.FILE }}" --json --excludePrivatePackages) >> $GITHUB_OUTPUT
+        run: echo DEPENDENCIES=$(npx license-checker-rseidelsohn@4.2.9 --direct --production --onlyAllow="${{ steps.allowed-licenses.outputs.FILE }}" --json --excludePrivatePackages) >> $GITHUB_OUTPUT
```
Comment (dzhavat) [inline .github/workflows/production-dependencies-license-check.yml:54]: question: didn't `@4` always take the latest on v4?

Comment (mimse) [reply .github/workflows/production-dependencies-license-check.yml:54]: Yes, but the latest version were broken :-(

File: libs/dh/wholesale/shell/src/lib/dh-wholesale-shell.routes.ts [hunk 4]
```diff
@@ -20,9 +20,19 @@ import { PermissionGuard } from '@energinet-datahub/dh/shared/feature-authorizat
...
   WHOLESALE_CALCULATIONS_PATH,
   WHOLESALE_SETTLEMENT_REPORTS_PATH,
+  WHOLESALE_REQUEST_CALCULATION_PATH,
 } from '@energinet-datahub/dh/wholesale/routing';
 
 export const dhWholesaleShellRoutes: Route[] = [
+  {
+    path: WHOLESALE_REQUEST_CALCULATION_PATH,
+    canActivate: [PermissionGuard(['settlement-reports:manage'])],
+    // What permission should this have? canActivate: [PermissionGuard(['calculations:manage'])],
```
[the remark this answers is not in the document]
Comment (mimse) [reply libs/dh/wholesale/shell/src/lib/dh-wholesale-shell.routes.ts:30]: Yes, still havent learned what permission they would like to have. But I´ll make a followup PR

**Review by dzhavat — APPROVED** [review]

**Pull request merged**